aries_cloudagent/core/plugin_registry.py

```diff
@@ -193,11 +193,12 @@ def register_package(self, package_name: str) -> Sequence[ModuleType]:
             )
         )
 
-    async def init_context(self, context: InjectionContext):
+    async def init_context(self, context: InjectionContext, plugins_config: dict = {}):
         """Call plugin setup methods on the current context."""
-        for plugin in self._plugins.values():
+        for key, plugin in self._plugins.items():
             if hasattr(plugin, "setup"):
-                await plugin.setup(context)
+                plugin_conf = plugins_config.get(key, {})
+                await plugin.setup(context, **plugin_conf)
```

Comment:
Altering this method signature does not seem like a good idea to me. The plugin configuration should be available to the plugin through the context. If the plugin wants to read config, it should elect to do so within the setup method.

Reply:
It has no impact on the setup method. I thought it was a clear and visible way to define the config variables. However, I agree with you that it has more sense if we use the context settings.

```
plugin_conf = plugins_config.get(key)
context.update_settings({"plugin_conf": plugin_conf})
await plugin.setup(context)
```